### ISSUE TITLE
Plugin Execution Configuration is not working

### ISSUE THREAD

**Issue by aliebelt** [javafx-maven-plugin/javafx-maven-plugin#82]
Hi,

we want to have different configurations for different executions of the javafx-maven-plugin. But it seems that the plugin doesn't use the execution configuration, e.g.:
```
			<plugin>
				<groupId>com.zenjava</groupId>
				<artifactId>javafx-maven-plugin</artifactId>
				<executions>
					<execution>
						<id>execution1</id>
						<phase>package</phase>
						<goals>
							<goal>native</goal>
						</goals>
						<configuration>
							<mainClass>my.MainApp1</mainClass>
						</configuration>
					</execution>
				</executions>
			</plugin>
```
Results in:
```
Failed to execute goal com.zenjava:javafx-maven-plugin:8.1.2:jar (jar) on project my-app:
The parameters 'mainClass' for goal com.zenjava:javafx-maven-plugin:8.1.2:jar are missing or invalid
```

**Comment by FibreFoX** [comment]
The problem are the used maven-mojo-annotation, with my PR 88 multiple executions should work.

**Comment by FibreFoX** [comment]
Just for information: i'm preparing to merge my fix in the next days. Multiple executions will be supported with the next version.

**Comment by maxd** [comment]
@FibreFoX  Seems like this PR bring problem. Look at this log of `mvn clean jfx:native` (without useless lines):

```
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ application ---
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ application ---
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ application ---
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ application ---
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ application ---
[INFO] --- maven-surefire-plugin:2.18.1:test (default-test) @ application ---
[INFO] --- maven-jar-plugin:2.6:jar (default-jar) @ application ---
[INFO] --- proguard-maven-plugin:2.0.9:proguard (default) @ application ---
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ application ---
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ application ---
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ application ---
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ application ---
[INFO] --- maven-surefire-plugin:2.18.1:test (default-test) @ application ---
[INFO] --- maven-jar-plugin:2.6:jar (default-jar) @ application ---
[INFO] --- proguard-maven-plugin:2.0.9:proguard (default) @ application ---
[INFO] --- javafx-maven-plugin:8.1.3-SNAPSHOT:jar (default-cli) @ application ---
[INFO] --- javafx-maven-plugin:8.1.3-SNAPSHOT:native (default-cli) @ application ---
```

As you can see the some build steps was executed twice (i.e. `proguard-maven-plugin` or `maven-compiler-plugin`). Use `git bisect` I have found a commit caused problem:

```
➜  javafx-maven-plugin git:(117e6f0) git bisect bad
117e6f0aabdca66d25785c03626df09ffd983375 is the first bad commit
commit 117e6f0aabdca66d25785c03626df09ffd983375
Author: FibreFoX <fibrefox@dynamicfiles.de>
Date:   Sat Jan 17 02:08:31 2015 +0100

    bugfix for issue #82 regarding multiple executions

    This fix introduces a new custom lifecycle called "jfxjar" and gives the possibility to have multiple executions.

:000000 100644 0000000000000000000000000000000000000000 a6f89c2da7a029afa02b6e7a2bf80ad34958a311 A	.gitignore
:040000 040000 d9a2e4d1055fed67fe2bdf18bc9995af419c0ecf df13e490bdb5a8750baf966a5da9b26a7f8a9846 M	src
```

Do you have any ideas how to fix it?

**Comment by FibreFoX** [comment]
Hm, how is your configuration of that project? ```NativeMojo``` and ```JarMojo``` are depending on jar:package goal, and because of multiple executions this would be called "fresh" for every execution. This is the correct way to depend on other phases, but maybe not as final as we wanting it to be ;)

You could fiddle with ```src/main/resources/META-INF/maven/lifecycle.xml``` to find a better way. I think its because both Mojos are depending on that phase, but multiple executions are fixed with it.

**Comment by maxd** [comment]
> Hm, how is your configuration of that project?

I don't use "advanced" configuration in my project. It is very similar to base, so you can reproduce problem use the following steps:

1. Create empty JavaFX project:

    `mvn archetype:generate -DarchetypeGroupId=com.zenjava -DarchetypeArtifactId=javafx-basic-archetype -DarchetypeVersion=2.0.1`

1. Build native bundle:

    `mvn jfx:native`

In my case the double execution caused problem with ProGuard. It shrink my JAR twice. + it increase build time.

How about revert this PR and try to fix issue without new life cycle? I don't think that need to use new life cycles to solve this issue. As workaround possible to use profiles to define different configurations.

**Comment by FibreFoX** [comment]
Hm, i will look into this a bit deeper, maybe i can find a solution fitting all requirements :cry: maybe i can find a solution also fitting to bug #65

**Comment by maxd** [comment]
ok, at present this problem block release of my project. How about revert this PR and create separate branch with your changes and possible solutions?

**Comment by FibreFoX** [comment]
You can revert my PR, i am working on my own fork anyway ;) (exactly for this kind of situations) i will create a new PR after finding a proper solution, would be nice, if you could validate that one. As a result, this issue remains open.

**Comment by maxd** [comment]
Yes, sure, I will help with validation this problem.

**Comment by FibreFoX** [comment]
I'm currently preparing a fix for this and some other issues ( #64 and #65 ), I hope to get this into my remote-branch the next two days, maybe a bit sooner. We are mixing some concepts here from mavens lifecycle and "standalone"-execution and it has to get corrected.
